src/qt/transactionfilterproxy.cpp

```diff
@@ -63,7 +63,7 @@ bool TransactionFilterProxy::filterAcceptsRow(int sourceRow, const QModelIndex&
     if (fOnlyZc && !isZcTx(type)){
         return false;
     }
-    if (fOnlyStakes && !isStakeTx(type))
+    if ((fOnlyStakes && !isStakeTx(type)) && (fShowOwnColdStakes && !isColdStakeMine(type)))
```

Comment:
This seems to work only as long as `fOnlyStakes` and `fShowOwnColdStakes` are either *both* true or *both* false (which is the case here).

But if, for example, we had `stakesFilter->setOnlyStakes(true, false)` or `stakesFilter->setOnlyStakes(false, true)`, then this condition would be **always** false (i.e. no record would be filtered out, entering the `if{}`):
```cpp
    if ((fOnlyStakes && !isStakeTx(type)) && (fShowOwnColdStakes && !isColdStakeMine(type)))
        return false;
```

I think the addition of `fShowOwnColdStakes` makes the logic needlessly complicated: why not simply keep the single parameter `fOnlyStakes`, and filter out those records that are't neither stake type nor coldstakemine type? 
e.g.
```cpp
    if (fOnlyStakes && !isStakeTx(type) && !isColdStakeMine(type)))
        return false;
```

Reply:
Hmm, i didn't stop much there, only the dashboard is using it anyway.

But agree that make things little bit more complicated. Easier to just add the `type == TransactionRecord::StakeDelegated` check inside `isStakeTx()` and done (Every regular stake must be from the wallet otherwise it wouldn't have been stored).